src/components/DataProvider/index.js

```diff
@@ -44,31 +43,59 @@ export default class DataProvider extends Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    const { series: newSeries, baseDomain: newDomain } = this.props;
+    const { series: oldSeries, baseDomain: oldDomain } = prevProps;
+
+    for (let i = 0; i < newSeries.length; i += 1) {
+      if (!oldSeries.find(s => s.id === newSeries[i].id)) {
+        return this.fetchData('SERIES_UPDATED');
+      }
+    }
+
+    for (let j = 0; j < oldSeries.length; j += 1) {
+      if (!newSeries.find(s => s.id === oldSeries[j].id)) {
```

Comment:
Existing series is deleted by default, the only thing we need to do is ask for data if there's a new serie coming in (even if it's toggled).

Reply:
If we had 2 series and now we have only 1 we need to update the chart somehow. Do we need just to filter seriesArray in that case?